webapp/src/Service/CheckConfigService.php

```diff
@@ -391,6 +392,33 @@ public function checkContestsValidate()
                     ($desc ?: 'No problems found.')];
     }
 
+    public function checkContestsDuration()
+    {
+        // Fetch all active and future contests
+        $contests = $this->dj->getCurrentContests(null, true);
+
+        $contestWarning = [];
+        $result = 'O';
+        foreach ($contests as $contest) {
+            $cid = $contest->getCid();
+            $duration = $contest->getEndtime()-$contest->getStarttime();
+            $multiplier = 60*60*24*7;
+            if ($duration/$multiplier>1) {
```

Comment:
Maybe `$this->getStartTimeObject()->add(new \DateInterval('P1W')) < $this->getEndTimeObject()` is more readable?

Reply:
I had to google this... but yes its more readable.